scapy/contrib/rtps/common_types.py

```diff
@@ -0,0 +1,323 @@
+            if hasattr(self.fld, "fmt"):
+                if len(self.fld.fmt) == 1:  # if it's only "I"
+                    _end = self.fld.fmt[0]
+                else:  # if it's "<I"
+                    _end = self.fld.fmt[1:]
+                self.fld.fmt = self.endianness + _end
+
+                self.fld.struct = struct.Struct(self.fld.fmt)
+
+    def getfield(self, pkt, buf):
+        self.set_endianness(pkt)
+        return self.fld.getfield(pkt, buf)
+
+    def addfield(self, pkt, buf, val):
+        self.set_endianness(pkt)
+        return self.fld.addfield(pkt, buf, val)
+
+    def randval(self):
+        return self.fld.randval()
+
+    def __getattr__(self, attr):
+        return getattr(self.fld, attr)
+
+
+class EPacket(Packet):
+    """A packet that manages its endianness"""
+
+    __slots__ = ["endianness"]
+
+    def __init__(self, *args, endianness=None, **kwargs):
+        self.endianness = endianness
+        super().__init__(*args, **kwargs)
+
+    def extract_padding(self, p):
+        return b"", p
+
+
+class EPacketField(PacketField):
+    """
+    A packet field that manages its endianness and that of its nested packet
+    """
+
+    __slots__ = ["endianness", "endianness_from", "fuzz_fun"]
+
+    def __init__(
+        self,
+        *args,
+        fuzz_fun=fuzz,
+        endianness=None,
+        endianness_from=e_flags,
+        **kwargs,
+    ):
+        self.endianness = endianness
+        self.endianness_from = endianness_from
+        self.fuzz_fun = fuzz_fun
+        super().__init__(*args, **kwargs)
+
+    def set_endianness(self, pkt):
+        if getattr(pkt, "endianness", None) is not None:
+            self.endianness = pkt.endianness
+        elif self.endianness_from is not None and pkt:
+            self.endianness = self.endianness_from(pkt)
+            assert self.endianness is not None
+
+    def m2i(self, pkt, m):
+        self.set_endianness(pkt)
+
+        _pkt = self.cls(m, endianness=self.endianness)
+
+        return _pkt
+
+    def randval(self):
+        if self.fuzz_fun is not None:
+            return self.fuzz_fun(self.cls())
+        return super().randval()
+
+
+class SerializedDataField(StrLenField):
+    pass
+
+
+class DataPacketField(EPacketField):
+    def m2i(self, pkt, m):
+        self.set_endianness(pkt)
+
+        pl_len = pkt.octetsToNextHeader - 24
+        _pkt = self.cls(
+            m,
+            endianness=self.endianness,
+            writer_entity_id_key=pkt.writerEntityIdKey,
+            writer_entity_id_kind=pkt.writerEntityIdKind,
+            pl_len=pl_len,
+        )
+
+        return _pkt
+
+
+class InlineQoSPacketField(EPacketField):
+    pass
+
+
+class PIDPadField(StrField):
+    def getfield(self, pkt, s):
+        len_pkt = 2  # TODO this is dynamic
+        return s[len_pkt:], self.m2i(pkt, s[:len_pkt])
+
+
+class FourBytesField(ByteField):
+    def __init__(self, name, default):
+        Field.__init__(self, name, default, ">I")
```

Comment:
This shoud be ByteField.__init__ (see https://app.codacy.com/gh/secdev/scapy/pullRequest?prid=8385225)

Reply:
They should probably extend `LEIntField` if the default format is `<I`

Reply:
I'm removing this and dependent classes as they're not used anymore.